scripts/rare-x/ingest_rarex_dataset.py

```diff
@@ -0,0 +1,189 @@
+    if not (job_status_code == 200 and job_status_response["job_status"] == "succeeded"):
+        print(f"{job_id} --> failed")
+        # if failed, get the resulting error message
+        job_result_code, job_result_response = get_job_result(job_id)
+        raise ValueError(job_result_response)    
+
+    # job completes successfully
+    # success_code is 200 and "job_status" = is "succeeded"
+    print(f"{job_id} --> succeeded")
+    
+    # get final job response for successfully completed ingest
+    job_result_code, job_result_response = get_job_result(job_id)
+
+    return job_result_response
+
+
+def ingest_dataset(dataset_id, data):
+    """Load data into TDR with ingestDataset."""
+
+    uri = f"https://data.terra.bio/api/repository/v1/datasets/{dataset_id}/ingest"
+
+    headers = {"Authorization": "Bearer " + get_access_token(),
+               "accept": "application/json",
+               "Content-Type": "application/json"}
+
+    response = requests.post(uri, headers=headers, data=data)
+    status_code = response.status_code
+
+    if status_code != 202:  # if ingest start-up fails
+        raise ValueError(response.text)
+
+    # if ingest start-up succeeds
+    return json.loads(response.text)
+
+
+def create_ingest_dataset_request(ingest_filepath, target_table_name):
+    """Create the ingestDataset request body."""
+
+    load_dict = {"format": "json",
+                 "path": ingest_filepath,
+                 "table": target_table_name,
+                 "resolve_existing_files": "true",
+                 "updateStrategy": "replace"
+                }
+
+    load_json = json.dumps(load_dict) # dict -> json
+
+    return load_json
+
+
+def run_ingest(dataset_id, ingest_data_json_filepaths):
+    """For each json path containing data to ingest, create ingest request and ingest to corresponding TDR dataset table."""
+
+    for path in ingest_data_json_filepaths:
+        # path = gs://bucket_name/subdir/filename.json
+        # filename = ingest destination table in dataset 
+        table_name = path.split("/")[-1].split(".")[0]
+        print(f"Starting ingest of {path} to table: {table_name}.")
+
+        # create request for ingestDataset
+        ingest_request = create_ingest_dataset_request(path, table_name) 
+        print(f"Submitted ingest request body: \n {ingest_request} \n")
+
+        # call ingestDatset
+        ingest_response = ingest_dataset(dataset_id, ingest_request) # call ingestDataset
+        print(f"Submitted ingest response body: \n {ingest_response} \n")
+
+        ingest_complete_response = monitor_ingest(ingest_response)
+        print(f"Finished ingest response body: \n {ingest_complete_response} \n")
+
+        print(f"Finished ingest for table {table_name}. \n\n")
+
+
+def get_ingest_data_json_filepaths(bucket_name, subdir):
+    """Get list of gs:// paths for each new-line delimited json containing data to ingest into TDR dataset table."""
```

Comment:
I think I'd specifically state the logic here as it relates to the input - you're looking for all json files in the given bucket_name/subdir